kin-core/src/main/java/kin/core/KinClient.java

```diff
@@ -125,7 +129,20 @@ KinAccount addAccount() throws CreateAccountException {
     KinAccount importAccount(@NonNull String exportedJson, @NonNull String passphrase)
         throws CryptoException, CreateAccountException, CorruptedDataException {
         KeyPair account = keyStore.importAccount(exportedJson, passphrase);
-        return addKeyPair(account);
+        KinAccount kinAccount = getAccountByPublicAddress(account.getAccountId());
+        return kinAccount != null ? kinAccount : addKeyPair(account);
+    }
+
+    @Nullable
+    private KinAccount getAccountByPublicAddress(String accountId) {
+        KinAccount kinAccount = null;
+        for (int i = 0; i < kinAccounts.size(); i++) {
+            final KinAccount account = kinAccounts.get(0);
```

Comment:
`0` -> `i`
add also test for multiple adding of accounts, that should be failed before fixing the `0` to `i`

Reply:
🥇